### DISCUSSION TITLE
Important: Remove, rename of branches, and new branch.

### DISCUSSION THREAD

**Discussion by Panchovix** [Panchovix/stable-diffusion-webui-reForge#175] [Announcements]
Hi there guys, hope you all are doing good.

I have made a lot of changes on the dev_upstream and dev_upstream_experimental branches, that both are gonna give issues if users had them before the changes of the past week.

The idea is at some point merge the changes of dev into main, which is not what is happening now.

So, **these branches are now removed and renamed**.

Their names now are:

dev_upstream -> dev
dev_upstream_experimental -> experimental

A new branch will be added, named:

dev2

Update 27 December 2024
How these branches are gonna work:

* dev: This branch has everything dev branch has, and all applicable updates from Comfy upstream, may have more features vs dev branch. Could have some more instabilities, but should also be generally stable for daily usage.
* dev2: Same as dev (for now)
* experimental: This branch will have some experimental changes that may have major new features, but they may be incomplete or have major bugs, based on the dev branch. This branch will be mostly inactive until I want to test things.

To use the new branches, you will need to fetch all changes and new branches, with:
`git fetch --all`

Then checkout what branch you want to use:

```
git checkout dev
# or
git checkout experimental
# or
git checkout dev2
```

If you were using the old branches (dev_upstream or dev_upstream_experimental), you should:

Make sure you don't have local changes you want to keep
Delete your local old branches:
```
git branch -D dev_upstream
git branch -D dev_upstream_experimental
```
Then checkout the new branch you want to use with the commands above.

**Note**: Simply doing 'git pull' on the old branches won't work as they have been removed from the repository. You must explicitly checkout the new branches.

Any question is welcome!

**Comment by Valentinbejan** [comment]
Dev_upstream used to have RAUNet/MSW-MSA for reForge but new in dev I don't have it. How can I get it back?

**Reply by Panchovix** [reply]
I moved out as a separated extension https://github.com/Panchovix/reforge_jankhidiffusion

**Reply by Valentinbejan** [reply]
Thanks, do I deactivate the built in one and install this one?

**Reply by Panchovix** [reply]
I suggest to delete the folder of the older one and the install this (to avoid duplicate folders and issues)

**Reply by Valentinbejan** [reply]
Thanks a lot :)
